pkg/clients/cloudsql/cloudsql.go

```diff
@@ -283,10 +296,17 @@ func LateInitializeSpec(spec *v1beta1.CloudSQLInstanceParameters, in sqladmin.Da
 
 // IsUpToDate checks whether current state is up-to-date compared to the given
 // set of parameters.
-func IsUpToDate(in *v1beta1.CloudSQLInstanceParameters, currentState sqladmin.DatabaseInstance) bool {
-	currentParams := &v1beta1.CloudSQLInstanceParameters{}
-	LateInitializeSpec(currentParams, currentState)
-	return cmp.Equal(in, currentParams, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}))
+func IsUpToDate(name string, in *v1beta1.CloudSQLInstanceParameters, currentState *sqladmin.DatabaseInstance) (bool, error) {
+	genState, err := copystructure.Copy(currentState)
```

Comment:
GCP unfortunately does not provide `DeepCopy` methods on their structs, so using a generic deep copy function here. We will pay a performance penalty, so it may be worth considering generating our own deep copy functions for them.